fbpic/openpmd_diag/boosted_particle_diag.py

```diff
@@ -459,13 +508,15 @@ def compact_slices( self, species ):
         -------
         paticle_array: an array of reals of shape (7, numPart)
         regardless of the dimension
-
-        Returns None if the slices are empty
+        Returns an array of size (7,0) if the slices are empty
```

Comment:
This modification is also made so that a proc which has no data at all can still participate in the MPI gather (by sending an empty array).